### ISSUE TITLE
FEAT: Replace codemirror with Monaco editor

### ISSUE THREAD

**Issue by geovannimp** [beekeeper-studio/beekeeper-studio#341]
**Is your feature request related to a problem? Please describe.**
Incomplete IntelliSense.

**Describe the solution you'd like**
Have a IDE like IntelliSense.

**Describe alternatives you've considered**
Replace codemirror with Monaco editor.

**Additional context**
I have been playing with monaco editor in the project and making some really amazing things.

Before open a PR I would like to know why codemirror has been chosen, things to check if still working with monaco and of couse if changing to monaco editor is a good choice or I'm totally crazy.

**Comment by rathboma** [comment]
Yes Monaco is much better :-) You are not crazy. I simply chose CodeMirror because I'd used it before for SQL editing and wanted to focus on other things.

But agreed Monaco is way more powerful.

**Comment by jenstornell** [comment]
Or wait for Codemirror 6 which is a complete rewrite. It's much faster and does not take up megabytes in filesize.

https://codemirror.net/6/

**Comment by rathboma** [comment]
I have no idea when code mirror 6 will be GA. There's no indication on the site.

**Comment by geovannimp** [comment]
@jenstornell Monaco editor should not be so heavy cause as I'm importing only sql module.

I'll make a todo at the PR to compare build size between actual version and refactored to monaco.

**Comment by rattrayalex** [comment]
Would love Monaco for more familiar editing (I use VS Code). It'd also be nice to have code folding enabled, if possible (I came to open a separate issue for that, but I assume it'd be better to wait until Monaco is the editor anyway).

**Comment by geovannimp** [comment]
@rattrayalex I stopped developing this case after a rebase the autocomplete just stopped working and I give up after some days trying to fix.

I don't have plans to work on this again any time soon, so I think a issue to add code folding to codemirror is welcome.

**Comment by rattrayalex** [comment]
OK, will do, thanks @geovannimp. Sorry to hear your work was not not salvageable.

**Comment by rathboma** [comment]
Moving editors is really tough. Also Monaco isn't what I'd call 'user friendly'. I think because it's extracted from VSCode rather than developed independently and merged in.

**Comment by rattrayalex** [comment]
Maybe this issue should be closed then?

**Comment by rathboma** [comment]
Maybe. Still noodling on it.
